src/core/socket.c

```diff
@@ -2323,6 +2345,56 @@ static void socket_enter_start_pre(Socket *s) {
                 socket_enter_start_open(s);
 }
 
+static bool socket_stop_notify(Unit *u) {
+        Socket *s = ASSERT_PTR(SOCKET(u));
+        _cleanup_(sd_bus_error_free) sd_bus_error error = SD_BUS_ERROR_NULL;
+        int r;
+
+        assert(s->state == SOCKET_DEFERRED);
+
+        r = manager_add_job(u->manager, JOB_START, UNIT_DEREF(s->service), JOB_MEEK, &error, /* ret = */ NULL);
+        if (r >= 0) { /* Yay! */
+                socket_set_state(s, SOCKET_RUNNING);
+                return true; /* changed */
+        }
+        if (sd_bus_error_has_name(&error, BUS_ERROR_TRANSACTION_IS_DESTRUCTIVE)) {
+                if (!hashmap_isempty(u->manager->jobs))
```

Comment:
Now max delay is implemented, then the condition should not be necessary anymore.